src/ngraph/runtime/tensor.cpp

```diff
@@ -70,3 +70,10 @@ void runtime::Tensor::set_stale(bool val)
 {
     m_stale = val;
 }
+
+void runtime::Tensor::copy_to(shared_ptr<runtime::Tensor> d, size_t offset, size_t n)
```

Comment:
The signature should be `copy_to(const runtime::Tensor&)` and no more. With this signature you can validate the call internally if you want and no need to pass offset and size as we know those from the tensors.
Don't pass shared_ptr around unless you really need to.